exec/cnex/lib/string.c

```diff
@@ -0,0 +1,143 @@
+    Cell *e = cell_newCellType(cString); // Temp element container for found strings.
+    size_t i = 0;
+    while (i < s->length) {
+        e->type = cString; // Ensure that we have the correct Cell type to operate on.
+        int64_t nd = string_findString(s, i, d);
+        if (nd == -1) {
+            e->string = string_subString(s, i, s->length - i);
+            cell_arrayAppendElement(r, *e);
+            cell_clearCell(e); // Since this calls free() on the string member, we're free to create a new string in this cell.
+            break;
+        } else if ((size_t)nd > i) {
+            e->string = string_subString(s, i, nd-i);
+            cell_arrayAppendElement(r, *e);
+            cell_clearCell(e); // Since this calls free() on the string member, we're free to create a new string in this cell.
+        }
+        i = nd + d->length;
+    }
+    cell_freeCell(e);
+    pop(exec->stack);
+    pop(exec->stack);
+
+    push(exec->stack, r);
+}
+
+void string_splitLines(TExecutor *exec)
+{
+    TString *s = peek(exec->stack, 0)->string; // Original String to split
+
+    Cell *r = cell_createArrayCell(0);
+    Cell *e = cell_newCellType(cString); // Temp element container for found strings.
+    size_t i = 0;
+    while (i < s->length) {
+        e->type = cString; // Ensure that we have the correct Cell type to operate on.
+        int64_t nl = string_findFirstOf(s, i, "\r\n");
+        if (nl == -1) {
+            e->string = string_subString(s, i, s->length - i);
+            cell_arrayAppendElement(r, *e);
+            cell_clearCell(e); // Since this calls free() on the string member, we're free to create a new string in this cell.
+            break;
+        }
+        e->string = string_subString(s, i, nl-i);
+        cell_arrayAppendElement(r, *e);
+        cell_clearCell(e); // Since this calls free() on the string member, we're free to create a new string in this cell.
+        if (s->data[nl] == '\r' && (size_t)nl+1 < s->length && s->data[nl+1] == '\n') {
+            i = nl + 2;
+        } else {
+            i = nl + 1;
+        }
+    }
+    cell_freeCell(e);
+    pop(exec->stack);
+
+    push(exec->stack, r);
+}
+
+void string_trimCharacters(TExecutor *exec)
+{
+    TString *ts= peek(exec->stack, 0)->string;  // Trailing Chars
+    TString *ls= peek(exec->stack, 1)->string;  // Leading Chars
+    TString *s = peek(exec->stack, 2)->string;  // String to trim
+
+    int64_t first = string_findFirstNotOf(s, ls);
+    int64_t last = string_findLastNotOf(s, ts);
+    if (first == -1 || last == -1) {
+        pop(exec->stack);
+        pop(exec->stack);
+        pop(exec->stack);
+        push(exec->stack, cell_createStringCell(0));
+        return;
+    }
+
+    Cell *r = cell_newCellType(cString);
+    // Note: string_subString() creates a new string; it does not modify the existing string in-place.
+    // ToDo: Would it be better to modify the string in place?  This should be tested and benchmarked.
```

Comment:
I don't understand this comment. This function returns a new string, modifying the original string in place doesn't make sense.

Reply:
And that's what the comment was sort of noting.  I wasn't sure if subString() should modify the source string, or create a new one.  I opted to create a new one.  I'll fix the comments so they make more sense.